src/CMakeLists.txt

```diff
@@ -26,11 +26,10 @@ configure_package_config_file(
     "${PROJECT_BINARY_DIR}/${CMAKE_FILES_DIRECTORY}/libappimageConfig.cmake"
     INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/libappimage
 )
-# ... for both
-configure_file(
-    "${PROJECT_SOURCE_DIR}/cmake/libappimageConfigVersion.cmake.in"
+write_basic_package_version_file(
     "${PROJECT_BINARY_DIR}/${CMAKE_FILES_DIRECTORY}/libappimageConfigVersion.cmake"
-    @ONLY
+    VERSION ${V_MAJOR}.${V_MINOR}.${V_PATCH}
+    COMPATIBILITY SameMajorVersion
```

Comment:
CMake 3.11 might be a too recent version to require perhaps for systems where libappimage should be buildable?
The other option would be to simply take the generated config code (assuming it works with older cmake on the target system) and for now keep a non-generated ConfigVersion file, just with 

(Sidenote; this discussion made me aware that most client dependency cmake code I have seen so far assumes that the provided version code only checks for the minimum version, and leaves any further if version >=/<= x.y details to the client, e.g, for API changes in minor versions.
Perhaps something to think about if that is an antipattern, or what asking for a min version actually means or should means, if we do not provide also a max version in the query)

Reply:
I doubt that we have to be compatible with anything but the latest versions of the distros with regards to the CMake version shipped there. In any non-packaging scenario, it's viable IMO to require the users to get a more recent CMake version.

Reply:
When it comes to balancing requirement of latest tools vs. supporting developers/potential contributors  out-of-the-box on more ancient systems, I rather opt to the latter.
But given my personal limited current needs with libappimage cmake 3.11 is fine, so totally relaxed in my chair seeing you raising this requirement :) You know your target audience,
I was just pointing this out as I am used to more legacy-supporting audience, so there was a small chance this version need challenges might have missed out on the radar when trying to find a solution to the problem here. :)

> Right now our compatibility criterion shouldn't be major but minor.

So far it seems to me 0.2 is ABI-backward-compatible to 0.1, no? Or is it just source-compatible?
Where do you see minor-backward-compatibility going to be broken?

Reply:
About backward ABI compatibility, we try to do our best by keeping the C interface but must of have been rewritten in C++. LibAppImage requires a C++ compiler and `Boost::FileSystem` now to properly build, but the code should remain unchanged. Haven't tried replacing libappimage 0.1 by 0.2 without recompiling, but I suppose it will be ok.

Reply:
"We try to do our best" is assumed by default ;)

Taking my libappimage consumer hat on, mainly I would like to know whether the ABI version _should_ be compatible (bugs always happen).
To be on the safe side, I would translate "I suppose it will be okay" into: "not backward-compatible", as this is different from a promise :)

When it comes to CMake Config files, when I as consumer are using those via find_package(libappimage), I am not interested how the binary product of libappimage has been created, so which tool and what ingredients are uninteresting, as long as they are not reflected in the public interface of the library my own software is going to use.

Reply:
We do require recompilation, the ABI will surely change. However, the API should remain the same. Whenever you notice changes there, that's an error and should be reported to us in a separate issue.